extensions/amp-story/1.0/amp-story.js

```diff
@@ -819,6 +826,20 @@ export class AmpStory extends AMP.BaseElement {
     return storyLayoutPromise;
   }
 
+  /**
+   * If the URL contains a valid page parameter value, the story's initial page
+   * should be the specified value.
+   * @param {string} initialPageId
+   * @return {string}
+   * @private
+   */
+  maybeChangeInitial_(initialPageId) {
```

Comment:
With these changes, the way we get the initial page ID is now done in two steps:
1. `const initialPageId = this.getHistoryState_(HistoryStates.PAGE_ID) || firstPageEl.id;`
2. Later, maybe changing it

What do you think of trying to do this in only one step? It might make things easier.
Like, where we get the `initialPageId` right now, we could have a `getInitialPageId_()` method, that would try to get the page ID in that order:
1. If there is a page ID in the history, use it (if a user refreshes their page, they'd expect to end up on the same page)
2. If there's a page ID in the URL fragment, use it
3. Fallback to the first page of the story

What do you think?

Reply:
This makes sense to me! I went ahead and made the changes. `maybeChangeInitial_` is now `getInitialPageId` and retrieves the correct pageId in the order you described. There's some code left because this is still behind an experiment wall but I mark that as a TODO to be removed explicitly.